src/Graphics/Image/Processing/Filter.hs

```diff
@@ -140,3 +141,78 @@ prewittOperator !img = sqrt (prewittX ^ (2 :: Int) + prewittY ^ (2 :: Int))
         !prewittY = applyFilter (prewittFilter Vertical Edge) img
 {-# INLINE prewittOperator #-}
 
+
+-- |The Laplacian of an image highlights regions of rapid intensity change 
+-- and is therefore often used for edge detection. It is often applied to an 
+-- image that has first been smoothed with something approximating a 
+-- Gaussian smoothing filter in order to reduce its sensitivity to noise.
+--
+-- <<images/yield.jpg>>   <<images/yield_laplacian.png>> 
+-- 
+-- Usage : 
+--	
+-- >>> img <- readImageY VU "images/yield.jpg"
+-- >>> let laplacianImage :: Image VU RGB Double
+-- >>>     laplacianImage = applyFilter (laplacianFilter Complete Edge) img
+-- >>> writeImage "images/yield_laplacian.png" (toImageRGB laplacianImage)
+--
+laplacianFilter :: (Array arr cs e, Array arr X e) =>
+                   Direction -> Border (Pixel cs e) -> Filter arr cs e
+laplacianFilter dir !border =
+  Filter (correlate border kernel)
+  where
+    !kernel =
+      case dir of
+        Complete   -> fromLists $ [ [ -1, -1, -1 ]     -- Unlike the Sobel edge detector, the Laplacian edge detector uses only one kernel. 
+                                  , [  -1, 8, -1 ]     -- It calculates second order derivatives in a single pass. 
+                                  , [  -1, -1, -1 ]]   -- This is the approximated kernel used for it. (Includes diagonals)
+{-# INLINE laplacianFilter #-}
+
+-- | 'Laplacian of Gaussian' (LOG) filter is a two step process of smoothing
+-- an image before applying some derivative filter on it. This comes in
+-- need for reducing the noise sensitivity while working with noisy
+-- datasets or in case of approximating second derivative measurements.
+-- 
+-- The LoG operator takes the second derivative of the image. Where the image  
+-- is basically uniform, the LoG will give zero. Wherever a change occurs, the LoG will
+-- give a positive response on the darker side and a negative response on the lighter side.
+-- 
+-- <<images/yield.jpg>>   <<images/yield_log.png>> 
+--
+logFilter :: (Array arr cs e, Array arr X e) =>
+             Direction -> Border (Pixel cs e) -> Filter arr cs e
+logFilter dir !border =
+  Filter (correlate border kernel)
+  where
+    !kernel =
+      case dir of
+        Vertical   -> fromLists $ [ [ 0, 1, 1, 2, 2, 2, 1, 1, 0 ]              
+                                  , [  1,  2,  4, 5, 5, 5, 4, 2, 1 ]
+                                  , [  1,  4,  5, 3, 0, 3, 5, 4, 1 ] 
+                                  , [  2,  5,  3, -12, -24, -12, 3, 5, 2 ] 
+                                  , [  2,  5,  0, -24, -40, -24, 0, 5, 2  ] 
+                                  , [  2,  5,  3, -12, -24, -12, 3, 5, 2  ] 
+                                  , [  1,  4,  5, 3, 0, 3, 5, 4, 1  ] 
+                                  , [  1,  2,  4, 5, 5, 5, 4, 2, 1  ] 
+                                  , [  0,  1,  1, 2, 2, 2, 1, 1, 0  ]] 
+
+        Horizontal -> fromLists $ [ [ 0, 1, 1, 2, 2, 2, 1, 1, 0 ]
```

Comment:
Kernel included twice. Just as with regular laplacian there is no sense of direction here, so there is only one kernel. It be better of it was something like that:
```haskell
  where
     !kernel = fromLists [ [ ...
```

Reply:
Oh ok. Yes sir, I'll change it.